app/include/zmk/studio/rpc.h

```diff
@@ -0,0 +1,209 @@
+ *        the pair or subsystem and request identifiers. This struct is the high level entity to
+ *        aggregate all the possible handler functions for the request in the given subsystem.
+ */
+struct zmk_rpc_subsystem {
+    subsystem_func *func;
+    uint16_t handlers_start_index;
+    uint16_t handlers_end_index;
+    uint8_t subsystem_choice;
+};
+
+/**
+ * @brief An entry for a specific handler function in a given subsystem, including metadata
+ * indicating if the particular handler requires the device be unlock in order to be invoked.
+ */
+struct zmk_rpc_subsystem_handler {
+    rpc_func *func;
+    uint8_t subsystem_choice;
+    uint8_t request_choice;
+    bool secured;
+};
+
+/**
+ * @brief Generate a "meta" subsystem response indicating an "empty" response to an RPC request.
+ */
+#define ZMK_RPC_NO_RESPONSE() ZMK_RPC_RESPONSE(meta, no_response, true)
+
+/**
+ * @brief Generate a "meta" subsystem response with one of a few possible simple error responses.
+ * @see   https://github.com/zmkfirmware/zmk-studio-messages/blob/main/proto/zmk/meta.proto#L5
+ */
+#define ZMK_RPC_SIMPLE_ERR(type)                                                                   \
+    ZMK_RPC_RESPONSE(meta, simple_error, zmk_meta_ErrorConditions_##type)
+
+/**
+ * @brief Register an RPC subsystem to aggregate handlers for request to that subsystem.
+ * @param prefix The identifier for the subsystem, e.g. `core`, `keymap`, etc.
+ * @see   https://github.com/zmkfirmware/zmk-studio-messages/blob/main/proto/zmk/studio.proto#L15
+ */
+#define ZMK_RPC_SUBSYSTEM(prefix)                                                                  \
+    zmk_studio_Response subsystem_func_##prefix(const struct zmk_rpc_subsystem *subsys,            \
+                                                const zmk_studio_Request *req) {                   \
+        uint8_t which_req = req->subsystem.prefix.which_request_type;                              \
+        return zmk_rpc_subsystem_delegate_to_subs(subsys, req, which_req);                         \
+    }                                                                                              \
+    STRUCT_SECTION_ITERABLE(zmk_rpc_subsystem, prefix##_subsystem) = {                             \
+        .func = subsystem_func_##prefix,                                                           \
+        .subsystem_choice = zmk_studio_Request_##prefix##_tag,                                     \
+    };
+
+/**
+ * @brief Register an RPC subsystem handler handler a specific request within the subsystem.
+ * @param prefix The identifier for the subsystem, e.g. `core`, `keymap`, etc.
+ * @param request_id The identifier for the request ID, e.g. `save_changes`.
+ * @param _secured Whether the handler requires the device be unlocked to allow invocation.
+ *
+ * @note  A function with a name matching the request_id must be in-scope and will be used as the
```

Comment:
Is this callback just a `void()` (seems unlikely)? The function signature or a typedef to a function pointer type would be useful here.